packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -137,6 +137,10 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
         return this.preferences[preferenceName] !== undefined;
     }
 
+    getPreferences(): { [key: string]: any } {
```

Comment:
Why is not defined in the `PreferenceService` interface?

Also, could you please point to a call site on the other branch to see how you intend to use it. Did you also consider to just add access to all keys instead of the map?

Reply:
That's here: https://github.com/theia-demo-plugins/wiptheia/pull/30/files#diff-0d38fc4e3d8d94838dc7490ae6216b4fR32

Reply:
@AlexTugarev I'll add it in the `PreferenceService` interface

Reply:
@mmorhun, thanks for sharing the call site. Please feel free to ask (on gitter or in GH issue) about the preference service. It looks like there the original preferences could be just used behind a proxy, no need to synchronize two states that way.

Reply:
@mmorhun and @akurinnoy, the reason I was asking about using key set instead of the underlying array was to avoid any interference with the preference service. Can you make it work by delegating to the preference service?

If copying is unavoidable, can you use a read-only clone for that?

Reply:
@AlexTugarev Thanks for review.

Reply:
@AlexTugarev @akurinnoy I created separate issue for this problem: https://github.com/theia-ide/theia/issues/2128.

Reply:
@mmorhun, the question on introducing an accessor for keys instead remains open, also if it's preferable to return a read-only copy instead.

Reply:
@akurinnoy, @mmorhun, how are you going to proceed with this? 
Can you comment on both suggestions? Either to add `getNames(): string[]` for all preference keys, or to change the implementation to return a readonly copy of preferences?